### ISSUE TITLE
Node layout and icon in Dynamo 2.13

### ISSUE THREAD

**Issue by erfajo** [DynamoDS/DynamoWishlist#205]
It is a bit problematic that you have chosen to make a circular icon on the nodes in the new 2.13 layout. I use the icon space also in the rectangular space. At the bottom of my icons have I added a horizontal bar in a specific color to indicate if the node is for projects, families, materials, or core nodes. This almost disappears in the new layout.

The icon as it should look like...
![Orchid Common DataStream Await Small](https://user-images.githubusercontent.com/2844374/149067228-8e9747bd-132e-4652-b251-535de2c81eed.png)

As you might have noticed have I added “namespace” to the info on mouse-over. This is just something I have added for my users, you don’t need to pay attention to that. However, I have also made a function to add “o|” in front of all my nodes when they are placed in the graph… that is an awful method, but it seems to be the only way we can indicate that nodes' origins from a non-dynamo team package. This should be solved otherwise, but we have asked for that in many years and nothing has happened, so I don’t presume anything is going to change :-S
![Snap001426](https://user-images.githubusercontent.com/2844374/149067566-a98f0a5e-2df0-4ab8-83e2-6e259162fb15.png)

My suggestion on the node layout is that the namespace should appear below the node name.
...and make the icon rectangular as original!
![Snap001428](https://user-images.githubusercontent.com/2844374/149092506-d382103f-fd58-4fc6-96e6-45a85faa57bb.png)

Also, I would find it useful if nodes with "default port values" could somehow be collapsed, and let the user decide to unfold a port to add a value feed. Furthermore, the arrows shown on a "port" should be used for something other than indicating a direction for data!?

Team layout in 2.13
![org layout](https://user-images.githubusercontent.com/2844374/149091219-76e6d7ae-9004-4275-b247-5e8888fd5ca5.png)

My suggestion
![colapse layout](https://user-images.githubusercontent.com/2844374/149092081-7f403828-ffab-4413-b8f9-bb81b1ae7057.png)

@Amoursol?

**Comment by Amoursol** [comment]
Hello @erfajo - Thank you for your thoughts! We are currently working through some UX designs around some of your concerns, namely;

- How to express this is a custom node
- How to customize this expression (i.e. your **o|**)

We're also exploring how to show host expression (i.e. Revit or Civil 3D), as well as active/inactive and function states more clearly. 

For clarity and consistency reasons, we will not be exploring the collapse node feature you suggested above. Do note that the chevron is a button to access port actions such as List@Level. 

FYI @Jingyi-Wen

**Comment by erfajo** [comment]
Sorry, just one thing more. With the new layout have you hidden your default icon for the nodes inside the xaml file… but why!? This is even worse, from my point of view, than package codes don’t have the option to have a default package icon!?

Besides, I didn't see any changes from circular to squared icons as asked for. You could easily have maintained using an icon with the white circle without any problems, instead of forcing all others to make new icons fit with your new layout. This is really not showing that the dynamo Team is listening to anyone else than yourselves. Therefore, I have retired all my icons and do now only have a handful of default icons, the work is simply too comprehensive :-(

**Comment by erfajo** [comment]
Since nothing is happening and I have given up that anything will happen, i close the issue...

**Comment by Amoursol** [comment]
This has now been introduced into Dynamo 2.19 onwards, manifesting in the "Zoom States". The nodes have a purple overlay and a package icon. 

![image](https://github.com/DynamoDS/DynamoWishlist/assets/12857357/713c2bed-787c-4c86-84b7-dee859173265)

**Comment by erfajo** [comment]
...in other words, we don't care about your suggestion.

**Comment by erfajo** [comment]
So just for the record.

Asking for a recognition of the package origin has many been asking for. If your answer to that is a colored layout when zooming, then I have to say “How is that a recognition of the package other than a generic solution not answering my solution”?. From my point of view, have you made something I have not asked for, or in other words it is not my suggestion you are answering, try to find the proper suggestion instead… if it exists.

Next, who has ever been asking for round icons? I am sure that the answer to that is “nobody!”, but someone in your team got you convinced that this was a great idea. However, the truth is, that it destroyed not only my icons but everybody's icons. I simply gave up and am now just using a few generic icons, I will not waste my time furthermore on that issue.
By the way, we are still not able to set a generic icon for either a package, a namespace, or a package yet… and as sad as it might be, we will presumable not ever be able to do that!

Lastly, collapsing default values is purely my idea, and I do understand that you don’t think this is a good idea, giving it a blue color on the port is a fair solution. 
But, I do have many nodes that take “Document” as input, and the same method (node) without the “Document” input, as in other cases of multiple method signatures. This gives double up on nodes for no purpose. So in the case of multiple signatures, it would be great if this somehow could be selected in the present node on the canvas. This was the greater idea behind my suggestion, but I failed to describe that.
However, until someone in your team comes up with the same idea, nothing will happen on that suggestion… and It will surely never happen that such an idea or any of my ideas ever will be looked upon, it will just happen due to some coincidence, it might even be a rather peripheral solution as the case with colored overlay when zooming!?

No It is not fair to say you are listening to your customers in general, but it is an Autodesk issue in general, no matter how you try to disguise it.
